### PR TITLE
Toggle submit

### PR DESCRIPTION
**Pull request by lc512k**
- dim the submit button when disabled
- trigger validation on change rather than focusout to validate checkbox
- trigger and listen for validation event on every field
- keep track of all required form fields to determine full form validation state (formValidity) to toggle submit button
- update date validation to be more robust
- add * to required field labels
- submit button is disabled by default when landing on form

cc/@ifyio

### CODE REVIEW

File: client/components/btn/main.scss [hunk 1]
```diff
@@ -8,6 +8,10 @@
 }
 
 .su-btn--primary {
-    background-color: #27757B;
+    background-color: rgb(39, 117, 123);
```
Comment (matthew-andrews) [inline client/components/btn/main.scss]: just a thought, has anyone taken you through our [colour palette](https://github.com/Financial-Times/o-colors)?  perhaps @keirog could spare a few minutes to talk through this :smile:

Comment (lc512k) [inline client/components/btn/main.scss]: Thanks @matthew-andrews! Had been looking for something like this, actually :)

File: views/signup.html [hunk 2]
```diff
@@ -319,7 +320,7 @@ <h1 class="su-heading">Subscribe to Premium Digital</h1>
 										</div>
 									</fieldset>
 
-									<button id="signUpSubmitButton" class="su-btn su-btn--primary su-ui-item--full-width js-form__submit-button">Complete subscription</button>
+									<button id="signUpSubmitButton" class="su-btn su-btn--primary su-ui-item--full-width js-form__submit-button" disabled>Complete subscription</button>
```
Comment (ifyio) [inline views/signup.html]: Just found out from @SarahHornsey (our designer) that the `o-button` component has styles for all the states, including the disabled state. So we can use that. It is here https://github.com/Financial-Times/next-signup/issues/77 . I have created an issue #77 to replace the current submit button with it. So we can get to it later (post MVP).

Comment (ifyio) [inline views/signup.html]: :+1:

**Pull request merged**